File: backend/app/models/chunk.py
# backend/app/models/chunk.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from pgvector.sqlalchemy import Vector
from app.core.database import Base

class CourseChunk(Base):
    __tablename__ = "course_chunks"

    id = Column(Integer, primary_key=True, index=True)
    course_id = Column(Integer, ForeignKey("courses.id"), nullable=False)
    section_id = Column(Integer, ForeignKey("sections.id"), nullable=True)
    module_title = Column(String)
    section_type = Column(String)
    text = Column(Text, nullable=False)
    embedding = Column(Vector(1536))

    course = relationship("Course", back_populates="chunks")

File: backend/src/models/Chunk.js
import mongoose from 'mongoose';

const chunkSchema = new mongoose.Schema(
  {
    courseId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Course',
      required: true,
      index: true,
    },
    moduleId: {
      type: String,
      default: '',
    },
    sectionId: {
      type: String,
      default: '',
    },
    text: {
      type: String,
      required: true,
    },
    embedding: {
      type: [Number],
      default: [],
    },
    metadata: {
      courseTitle: { type: String, default: '' },
      moduleTitle: { type: String, default: '' },
      sectionTitle: { type: String, default: '' },
      sectionType: { type: String, default: 'video' },
      startTime: { type: Number, default: 0 },
      endTime: { type: Number, default: 0 },
    },
  },
  {
    timestamps: true,
  }
);

export const Chunk = mongoose.model('Chunk', chunkSchema);
